changes/bug40062

```diff
@@ -0,0 +1,5 @@
+  o Major bugfixes (onion services):
+    - When writing an onion service hostname file, first read it to make
+      sure it contains what we want before attempting to write it. This
+      prevents a bug where we can't read an onion service directory if it's
+      read-only. Fixes bug 40062; bugfix on 0.0.6. Patch by Neel Chauhan.
```

Comment:
I'd actually call this a 'Minor feature', since the behavior didn't exist before, and now it does. And then change 'This prevents a bug where' to something like 'Now onion services can set their existing onion service directories to read-only and Tor will still work'. I think the convention for features is to say "Resolves ticket x" rather than "Fixes bug x".

I'd also suggest squashing the "Use write_str_if_not_equal() for v2 onion services" and "Use write_str_if_not_equal() for v3 onion services" commits into this one, so ultimately there is one commit that refactors the function, and one commit that makes the two new changes plus adds the changes file.

Having the two behavior changes, plus the changes file, in a single commit might help somebody in the distant future who is trying to figure out what changed, and they see one commit but they don't realize there is a nearby very similar one.

Thanks!